Packs/ProofpointEmailSecurity/Integrations/ProofpointEmailSecurityEventCollector/ProofpointEmailSecurityEventCollector_test.py

```diff
@@ -0,0 +1,89 @@
+import uuid
+import pytest
+from ProofpointEmailSecurityEventCollector import fetch_events, json, demisto, EventType, datetime, timedelta
+import ProofpointEmailSecurityEventCollector
+from freezegun import freeze_time
+
+CURRENT_TIME: datetime | None = None
+
+
+def is_interval_passed(fetch_start_time: datetime, fetch_interval: int) -> bool:
+    global CURRENT_TIME
+    if not CURRENT_TIME:
+        CURRENT_TIME = fetch_start_time
+    return fetch_start_time + timedelta(seconds=fetch_interval) < CURRENT_TIME
+
+
+@pytest.fixture
+def connection():
+    # Set up a mock connection
+    return MockConnection()
+
+
+class MockConnection:
+    def __init__(
+        self,
+    ):
+        global CURRENT_TIME
+        self.id = uuid.uuid4()
+        self.events = [
+            {"ts": "2023-08-16T13:24:12.147573+0100", "message": "Test message 1", "id": 1},
+            {"ts": "2023-08-14T13:24:12.147573+0200", "message": "Test message 2", "id": 2},
+            {"ts": "2023-08-12T13:24:11.147573+0000", "message": "Test message 3", "guid": 3},
+        ]
+        self.index = 0
+
+    def recv(self, timeout):
+        global CURRENT_TIME
+        # pretend to sleep for 4 seconds
+        assert CURRENT_TIME
+        CURRENT_TIME += timedelta(seconds=4)
+
+        if self.index >= len(self.events):
+            raise TimeoutError
+        event = self.events[self.index]
+        self.index += 1
+        return json.dumps(event)
+
+
+def test_fetch_events(mocker, connection):
```

Comment:
looks bit short. what is the coverage?

Reply:
Added more tests. Coverage now is over 75 percent now